src/lib.rs

```diff
@@ -667,6 +669,34 @@ decl_module! {
 
             Ok(())
         }
+
+        #[weight = FunctionOf(
+            |args: (&T::AccountId, &u8, &Timepoint<T::BlockNumber>)| 500_000 * (*args.1 as u32),
+            DispatchClass::Normal,
+            true
+        )]
+		fn clear(origin, multi_account_id: T::AccountId, max_cancellations: u8, timepoint: Timepoint<T::BlockNumber>) -> DispatchResult {
+			let mut succeeded = 0;
+			let who = ensure_signed(origin)?;
+			let mut cancelled_multisigs = Vec::with_capacity(max_cancellations as usize);
+			let active_multisigs = <Multisigs<T>>::iter_prefix(&who);
+			for m_sig in active_multisigs.take(max_cancellations as usize){
+				ensure!(who == multi_account_id || who == m_sig.depositor, Error::<T>::NotOwner);
+
+				let _ = T::Currency::unreserve(&m_sig.depositor, m_sig.deposit);
```

Comment:
This check must not happen in the same loop than can error - in the current implementation a later error can abort the cancelling, but earlier deposits would have been refunded. Please use multiple loops.

Reply:
@philipstanislaus what should be the check here?
```rust
ensure!(who == multi_account_id || who == m_sig.depositor, Error::<T>::NotOwner);
```
This check will make the whole function fail, since each mulitsig account will be different from the origin/caller.

Should the check be any member of the multi-account be able to clear the multisig?

Reply:
No, only the multi_account_id itself and the depositor should be able to cancel a multisig.